src/vs/base/parts/tree/browser/treeModel.ts

```diff
@@ -960,6 +960,9 @@ export class TreeModel {
 	}
 
 	public refresh(element: any = null, recursive: boolean = true): Promise<any> {
+		let self = this;
+		// Add loading spinner for item during refresh process.
+		this.addTraits('loading', [element]);
```

Comment:
So any changes in the VS folder should always have a SQL CARBON EDIT. But in this case this stuff is a super old version of the tree that VS removed a long time back (https://github.com/microsoft/vscode/commit/f28dbf7cfc24c876dae2b006d3b3b381b7b722f9) so this is fine, although I think I'll go and actually move this into the sql folder then to reduce the "extra" stuff we have in the VS folder. 

(this is more of a general comment than anything you need to do anything about)

Reply:
Didn't realize it was in the vs folder until now.